### PR TITLE
RST Conversion: Overview/*

### PR DESCRIPTION
**Pull request by shieldsjared**
`faq.md` -> `faq.rst`
- One-to-one conversion of content
- NOTE: page is not linked in the TOC anywhere (e.g. orphaned content).  Not sure if this is deliberate or not.

`overview.md` -> `overview.rst`
- Page is not linked in the TOC
- Content is mirrored in `product.rst` (which is linked)
- Replaced content with encouragement for the user to see `product.rst` for updated content.

### CODE REVIEW

File: source/overview/faq.rst [hunk 1]
```diff
@@ -0,0 +1,37 @@
+Frequently Asked Questions 
+==========================
+
+General Questions 
+-----------------
+
+Why was Mattermost created?
+    Mattermost was created to offer an alternative to propreitary SaaS services. For more info, please go to mattermost.org and click "Why we made Mattermost".
+
```
Comment (JeffSchering) [inline source/overview/faq.rst]: Mattermost.org is not available for now. See https://pre-release.mattermost.com/core/pl/wnwfwp11i3b9icrtpdsqjjtfgc

You could use a direct link to https://www.mattermost.org/why-we-made-mattermost-an-open-source-slack-alternative/ but there's no knowing how long that link will last.

Maybe a paragraph or two summary of the "Why we made Mattermost" page?

File: source/overview/faq.rst [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+Will the open source Team Edition offer more authentication add-ons?
+    There are no current plans for auth add-ons for the following reasons: 
+    1. Authentication is a critical path that needs to be tested, maintained and supported with each bi-monthly release and volunteers aren't prepared to bear the tax this change would impose on them, and
+    2. Such a change increases the difficulty of creating high quality apps and drivers, as developers would be taxed with requests for support, troubleshooting, and debug set ups that would be expensive to reproduce.
+
```
Comment (JeffSchering) [inline source/overview/faq.rst]: 'set ups' should be 'setups'. Use *setup* as a noun, and *set up* as a verb.

File: source/overview/faq.rst [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+Why does the open source repo contain code specific to the commercial version of Mattermost?
+    The commercial versionw of Mattermost is designed to never lock-in your data. Portions of the commercial version are shared with the open source version to ensure upgrade and downgrade across editions happens without data loss. 
+
```
Comment (JeffSchering) [inline source/overview/faq.rst]: Typo - 'The commercial versionw ...' should be 'The commercial version ...'

File: source/overview/faq.rst [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    **Functional Scaling** - Scaling from a team to an enterprise is like going from a "virtual office" to a "virtual campus". Advanced features like enterprise authentication, granular permissions, compliance and auditing, and advanced reporting become increasingly important as organizations grow beyond teams. Organizations needing this flexibility can easily upgrade from Mattermost Team Edition to Mattermost Enterprise Edition, as well as downgrade without data loss, should their needs change. 
+
+    For more information on how Mattermost scales, technically and functionality, please `contact the Enterprise team <https://about.mattermost.com/contact/>`_.
```
Comment (JeffSchering) [inline source/overview/faq.rst]: 'functionality' should be 'functionally'

**Comment by jasonblais** [comment]
Thanks @shieldsjared and @JeffSchering! 

Made proposed tweaks (I pushed them to the branch since they were small, hope that's okay @shieldsjared).

The MD --> RST looks okay, but it worthwhile revisiting the FAQ doc at some point, and add it back to the TOC after the updates.

**Comment by shieldsjared** [comment]
I never mind collaboration @jasonblais! Thx for knocking them out. I didn't even thing to actually _read_ the content, LOL.

FAQ - agreed on TOC. Not sure if it was abandoned intentionally or not.

**Comment by esethna** [comment]
Thanks @shieldsjared! I committed some minor formatting tweaks, other than that the conversion looks great, thanks!

**Pull request merged**